x-pack/legacy/plugins/ml/public/util/custom_url_utils.ts

```diff
@@ -83,67 +91,86 @@ export function openCustomUrlWindow(fullUrl, urlConfig) {
-
-    // If the token is an influencer, then the value in the record will be an array.
-    // For now just support passing the first influencer from the array.
-    // TODO - support passing multiple influencer values.
-    if (Array.isArray(tokenValue)) {
-      tokenValue = tokenValue[0];
-    }
-
-    if (tokenValue !== null && !(name === 'earliest' || name === 'latest')) {
-
-      if (urlValue.includes('language:lucene') === true) {
-        // Escape reserved characters if the query language is lucene (default was switched to KQL in 7.1).
-        tokenValue = `${escapeForElasticsearchQuery(tokenValue)}`;
-      } else {
-        // Escape any double quotes in the value for correct use in KQL.
-        tokenValue = tokenValue.replace(/\"/g, '\\"');
-      }
-
-      // Kibana URLs used rison encoding, so escape with ! any ! or ' characters
-      tokenValue = tokenValue.replace(/[!']/g, '!$&');
-
-      // URI encode in case of special characters in the value.
-      tokenValue = encodeURIComponent(tokenValue);
-    }
-
-    // If property not found string is not replaced.
-    return (tokenValue !== null) ? tokenValue : match;
-  });
+  const queryLanguageEscapeCallback = urlValue.includes('language:lucene')
+    ? escapeForElasticsearchQuery
+    : escapeForKQL;
+
+  // Compose callback for characters escaping and encoding.
+  const getResultTokenValue: (v: string) => string = _.flow(
+    queryLanguageEscapeCallback,
+    // Kibana URLs used rison encoding, so escape with ! any ! or ' characters
+    (v: string): string => v.replace(/[!']/g, '!$&'),
+    encodeURIComponent
+  );
+
+  return String(urlValue)
+    .replace('$earliest$', record.earliest)
+    .replace('$latest$', record.latest)
+    .replace(/query:'(.+)'/, (match, queryString: string) => {
+      return `query:'${queryString
+        // Split query string by AND operator.
+        .split(/\sand\s/i)
+        // Get property name from `influencerField:$influencerField$` string.
+        .map(v => v.split(':')[0])
+        .map(name => {
+          // Use lodash get to allow nested JSON fields to be retrieved.
+          const tokenValues: string[] = _.get(record, name) || null;
+          if (tokenValues === null) {
+            return null;
+          }
+          // Create a pair ``influencerField:value`.
+          // In cases where there are multiple influencer field values for an anomaly
+          // combine values with OR operator e.g. `influencerField:value or influencerField:another_value`.
+          const result = tokenValues
+            .map(value => `${name}:"${getResultTokenValue(value)}"`)
+            .join(' or ');
+          return tokenValues.length > 1 ? `(${result})` : result;
+        })
+        .filter(v => v !== null)
+        .join(' and ')}'`;
+    })
+    .replace(/\$(\w+)\$/, (match, name: string) => {
+      // Use lodash get to allow nested JSON fields to be retrieved.
+      const tokenValue: string | undefined = _.get(record, name);
```

Comment:
Nit: This could use `elastic-idx` instead of lodash's `get`.

Reply:
as we discussed `getNestedProperty()` from `x-pack/legacy/plugins/ml/public/util/object_utils.ts` is not going to work here due to accessor in variable